src/sentry/tagstore/v2/backend.py

```diff
@@ -78,32 +66,46 @@ def setup(self):
             grouptagvalue_model=GroupTagValue,
         )
 
+    def setup_deletions(self, **kwargs):
+        from sentry.deletions import default_manager as deletion_manager
+        from sentry.deletions.defaults import BulkModelDeletionTask
+        from sentry.deletions.base import ModelRelation
+        from sentry.models import Project
+
+        deletion_manager.register(TagKey, BulkModelDeletionTask)
+        deletion_manager.add_dependencies(Project, [
+            lambda instance: ModelRelation(TagKey, {'project_id': instance.id}),
+        ])
+
+        super(V2TagStorage, self).setup_deletions(**kwargs)
```

Comment:
Same comments here with regard to ordering of these `super` calls as the other backend.